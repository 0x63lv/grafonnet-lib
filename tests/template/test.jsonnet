local grafana = import "grafonnet/grafana.libsonnet";
local template = grafana.template;

{
    basic: template.new("host", "elastic", "test-query"),
    advanced: template.new(
        "host",
        "elastic",
        "test-query",
        label="node",
        allValues="$ALL",
        tagValuesQuery="tvquery",
        current={ foo: "bar" },
        hide="value",
        includeAll=true,
        multi=true,
    ),
}
