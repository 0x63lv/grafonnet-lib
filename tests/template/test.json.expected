{
   "advanced": {
      "allValue": "$ALL",
      "current": {
         "foo": "bar"
      },
      "datasource": "elastic",
      "hide": 2,
      "includeAll": true,
      "label": "node",
      "multi": true,
      "name": "name",
      "options": [ ],
      "query": "TEST",
      "refresh": 0,
      "regex": "",
      "sort": 0,
      "tagValuesQuery": "tvquery",
      "tags": [ ],
      "tagsQuery": "",
      "type": "query",
      "useTags": false
   },
   "basic": {
      "allValue": null,
      "current": { },
      "datasource": "elastic",
      "hide": 0,
      "includeAll": false,
      "label": null,
      "multi": false,
      "name": "name",
      "options": [ ],
      "query": "TEST",
      "refresh": 0,
      "regex": "",
      "sort": 0,
      "tagValuesQuery": "",
      "tags": [ ],
      "tagsQuery": "",
      "type": "query",
      "useTags": false
   }
}
